client/src/launchGdcMatrix.js

```diff
@@ -1,31 +1,43 @@
 import { dofetch3 } from '#common/dofetch'
-import { make_one_checkbox } from '#dom/checkbox'
 import { appInit } from '#plots/plot.app.js'
 import { fillTermWrapper } from '#termsetting'
 
 /*
-
 test with http://localhost:3000/example.gdc.matrix.html
 runpp({ launchGdcMatrix:true })
 
-designed to work for Oncoprint app in GDC Analysis Tools Framework
+designed to work for OncoMatrix app in GDC Analysis Tools Framework
 
 initiates a "plot" app instance that wraps around the matrix component
 and returns the plot app API
 
 
 ********* parameters
 
+holder
+genomes = { hg38 : {} }
 arg = {}
 	runpp() argument object
 
 	.filter0{}
-		the filter object from GDC Cohort Builder
+		the filter object from GDC Cohort Builder, for restricting cases in oncomatrix
 		{"op":"and","content":[{"op":"in","content":{"field":"cases.primary_site","value":["breast","bronchus and lung"]}}]}
+		not visualized on pp (invisible) and read only, passed to backend -> gdc api
+		// NOTE 6/29/2023: this is the default only in the OncoMatrix demo mode,
+		// and is supplied by the OncoMatrixWrapper in the GFF repo, no need to supply here
+		// return { op: 'in', content: { field: 'cases.disease_type', value: ['Gliomas'] } }
 
 	.genes[]
 		future: list of genes to launch matrix
 
+	.termgroups[]
+		TODO 
+	
+	.settings{}
```

Comment:
@siosonel could you comment on the reason for needing two attributes `arg.settings.matrix{}, arg.opts.matrix{}`

Reply:
@xzhou82 
1. `arg.settings.matrix`: any state that can be JSON-encoded/hydrated as part of `getPlotConfig().settings.matrix`. This is meant to hold plot-instance-specific state that only affects rendering and not data requests, but that distinction is not as clear with plots that use server-side rendering like violin plot.

2. `arg.plots.matrix`:  any options that cannot be JSON-stringified like callbacks. These options will also be applied to all rehydrated or new plots of the same type, so not plot-instance-specific.